Mathlib/CategoryTheory/Limits/Skeleton.lean

```diff
@@ -0,0 +1,227 @@
+/-
+Copyright (c) 2023 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.CategoryTheory.Skeletal
+import Mathlib.CategoryTheory.Limits.Shapes.Biproducts
+import Mathlib.CategoryTheory.Limits.Preserves.Shapes.Biproducts
+
+/-!
+# The additive monoid on the skeleton of a category with binary biproducts
+
+## Main results:
+
+* `Skeleton.instAddCommMonoid`
+
+-/
+
+namespace CategoryTheory
+
+universe v u v' u'
+
+section Skeletal
+
+variable {C : Type u} [Category.{v} C]
+variable [Limits.HasZeroMorphisms C] [Limits.HasBinaryBiproducts C] [Limits.HasZeroObject C]
+
+/-- If `C` has binary biproducts and is skeletal, it is a commutative additive monoid.
+See note [reducible non-instances]. -/
+@[reducible]
+noncomputable def addCommMonoidOfSkeletal (hC : Skeletal C) : AddCommMonoid C where
+  add X Y := (X ⊞ Y : C)
+  toZero := CategoryTheory.Limits.HasZeroObject.zero' _
+  zero_add X := hC ⟨(Limits.isoZeroBiprod (Limits.isZero_zero _)).symm⟩
+  add_zero X := hC ⟨(Limits.isoBiprodZero (Limits.isZero_zero _)).symm⟩
+  add_assoc X Y Z := hC ⟨Limits.biprod.associator X Y Z⟩
+  add_comm X Y := hC ⟨Limits.biprod.braiding X Y⟩
+
+end Skeletal
+
+/-! TODO: surely we have all this stuff somewhere? -/
```

Comment:
[Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/transporting.20categorical.20limits.20across.20equivalences/near/421420611)